tracer/src/Datadog.Trace/Tagging/LazyOrString.cs

```diff
@@ -0,0 +1,29 @@
+﻿// <copyright file="LazyOrString.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+
+namespace Datadog.Trace.Tagging
+{
+    internal class LazyOrString
```

Comment:
Is this class needed? Could you use Lazy<string> directly?

Reply:
I think the idea is to avoid the `Func<string>` overhead (delegate allocation and invocation) if not needed. `Lazy<T>` support this since .NET Core 2.0, but .NET Framework does not.

I do wonder if we can make this a `readonly struct` to avoid yet another allocation when possible.